src/servers/grpc_server.cc

```diff
@@ -3291,6 +3291,27 @@ ModelInferHandler::Process(InferHandler::State* state, bool rpc_ok)
   return !finished;
 }
 
+TRITONSERVER_Error* GetEncodedError(TRITONSERVER_InferenceResponse* res_, TRITONSERVER_Error* err_)
+{
+    // since grpc drops msg if status is not ok,
+    // attach request id to the error message
+    // grpc_client needs to parse it appropriately
+    const char *id;
+    LOG_TRITONSERVER_ERROR(
+      TRITONSERVER_InferenceResponseId(res_, &id),
+      "couldn't retrieve id for failed request");
+    TRITONSERVER_Error_Code err_code = TRITONSERVER_ErrorCode(err_);
+    std::string encoded_msg(TRITONSERVER_ErrorMessage(err_));
+    encoded_msg.append(" request_id = # ");
```

Comment:
The request_id field is optional in request message. If id is empty, then the `request_id = #` should not be attached to the error message. See my comment below. You should not need to add such encoded messages.